utils/retryexecutor.go

```diff
@@ -63,5 +69,15 @@ func (runner *RetryExecutor) LogRetry(attemptNumber int, err error) {
 	} else {
 		log.Debug(message)
 	}
+}
 
+func (runner *RetryExecutor) isCancelled() bool {
```

Comment:
I suggest having this function return the context cancellation error instead of a boolean. That would allow the retry executor to return the actual error that caused it stop.
I therefoer also suggest renaming this function to `checkCancelled`.